android-core/src/main/java/com/mparticle/modernization/identity/example/MParticleIdentityImpl.kt

```diff
@@ -0,0 +1,60 @@
+package com.mparticle.modernization.identity.example
+
+import com.mparticle.identity.IdentityApiRequest
+import com.mparticle.identity.IdentityApiResult
+import com.mparticle.identity.MParticleUser
+import com.mparticle.modernization.MParticleCallback
+import com.mparticle.modernization.core.MParticleMediator
+import com.mparticle.modernization.identity.IdentityCallback
+import com.mparticle.modernization.identity.InternalIdentity
+
+internal class MParticleIdentityImpl(private val mediator: MParticleMediator) :
```

Comment:
Is this introducing new behavior where Identity can be blocking?  Or is this the plan for future state?

Reply:
It looks like these are blocking but can only be used within a coroutine which are actually async. So I think this is just another way of doing async code. Mark can provide more context on the specifics of coroutine usage, but I believe these are still meant to ultimately be used asynchronously.

Reply:
Exactly, my ultimate idea is to do everything that makes sense and more performant to be done async, managed by the single coroutine thread pool in the Mediator. Because the implementation "XYZImpl" should be internal, it makes sense to do all the internal with a single way of handling async (threads and runnables, coroutines, etc.) We can then expose, wrap and provide further functionality in the external apis if needed.